mesonbuild/compilers/compilers.py

```diff
@@ -1288,9 +1288,10 @@ def __init__(self, gcc_type, defines):
                              'b_ndebug', 'b_staticpic',
                              'b_lundef', 'b_asneeded']
 
-    def get_library_dirs(self):
-        env = os.environ.copy()
-        env['LC_ALL'] = 'C'
+    def get_library_dirs(self, env=None):
+        if env is None:
+            env = os.environ.copy()
+            env['LC_ALL'] = 'C'
         stdo = Popen_safe(self.exelist + ['--print-search-dirs'], env=env)[1]
```

Comment:
You will want to change this to also use `_build_wrapper()` and the `get_program_dirs()` below too.

Reply:
Ok.